pkg/scheduler/eventhandlers.go

```diff
@@ -242,8 +242,15 @@ func assignedPod(pod *v1.Pod) bool {
 }
 
 // responsibleForPod returns true if the pod has asked to be scheduled by the given scheduler.
-func responsibleForPod(pod *v1.Pod, profiles profile.Map) bool {
-	return profiles.HandlesSchedulerName(pod.Spec.SchedulerName)
+func responsibleForPod(pod *v1.Pod, profiles profile.Map, recorderFactory profile.RecorderFactory) bool {
+	isResponsible := profiles.HandlesSchedulerName(pod.Spec.SchedulerName)
+	if !isResponsible && pod.DeletionTimestamp == nil {
+		// Prepare event clients.
+		recorder := recorderFactory("kube-scheduler")
+		recorder.Eventf(pod, nil, v1.EventTypeWarning, "FailedScheduling", "Scheduling",
+			"%v scheduler not found ", pod.Spec.SchedulerName)
+	}
+	return isResponsible
```

Comment:
We cannot go with this implementation. This scheduler doesn't match with `SchedulerName`, it doesn't mean the cluster doesn't have the scheduler matched with `SchedulerName`
Other schedulers may be working in another Pod/process.

Reply:
we need notify the SchedulerName not in kube-scheduler regardless of whether other schedulers are deployed, as for other scheduler(kube-batch、volcano) we can do nothing.

Reply:
> regardless of whether other schedulers are deployed

I don't agree. `SchedulerName` is the API to define which scheduler is responsible for the Pod, and if `SchedulerName` doesn't match with this scheduler, ignores that Pod: this behavior does completely make sense. 
If we record `FailedScheduling` here, how another scheduler (matching with `SchedulerName`) could work for this Pod?

Reply:
maybe the `FailedScheduling` maybe confuse，i chage the content of event message. as fellow

```
0s          Warning   FindScheduler      pod/web-0         test scheduler not in [default-scheduler]
```

Reply:
This will cause multiple warning events if there are multiple schedulers, and actually someone scheduler match the given `SchedulerName`

Reply:
Again, I would say we cannot go with this. As @lowang-bh said, if multiple schedulers are there, all Pods will get the warning events **even if everything works well.**

Reply:
I agree with @sanposhiho, we shouldn't record an event because the even stream is shared between all controllers handling the pod, and so this will be confusing because the assumption here is that the other scheduler will pick up the pod and handle it. Also, here you are recording an event on each update, which will spam the event stream.

I would be onboard with a log that gets printed on add only (this will probably require adding a dedicated event handler though).

Reply:
> I agree with @sanposhiho, we shouldn't record an event because the even stream is shared between all controllers handling the pod, and so this will be confusing because the assumption here is that the other scheduler will pick up the pod and handle it. Also, here you are recording an event on each update, which will spam the event stream.
> 
i consider the scenario where the update event occurs，because the `&&` operator,  don't record a event when pod was scheduled.

Reply:
In the scenarios I encountered, there were few custom schedulers based on standard scheduler implementations, more like [kube-batch](https://github.com/kubernetes-sigs/kube-batch)、[hived](https://github.com/microsoft/hivedscheduler)、[volcano](https://github.com/volcano-sh/volcano).  

Do you have any good methods？let me try